dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/icon/IconControllerTest.java

```diff
@@ -126,6 +121,19 @@ void shouldDeleteIconWhenKeyExists() throws IOException {
     assertEquals(String.format("Icon %s deleted", iconKey), response.getString("message").string());
   }
 
+  @Test
+  void shouldGetOnlyCustomIcons() throws IOException {
+    String fileResourceId = createFileResource();
+    createIcon(fileResourceId);
+
+    JsonArray iconArray = GET("/icons/type/custom").content(HttpStatus.OK);
+
+    assertEquals(iconKey, iconArray.getObject(0).getString("key").string());
+    assertEquals(description, iconArray.getObject(0).getString("description").string());
+    assertEquals(fileResourceId, iconArray.getObject(0).getString("fileResourceUid").string());
+    assertEquals(keywords, iconArray.getObject(0).getArray("keywords").toString());
+  }
```

Comment:
It would be nice to wrap this kind of assertion in `assertAll()`. Also, with an error message in case it fails

Reply:
fixed